src/lib.rs

```diff
@@ -90,10 +90,13 @@ mod ser;
 
 pub mod binary_heap;
 pub mod i;
-#[cfg(not(armv6m))]
+#[cfg(feature = "cas")]
 pub mod mpmc;
-#[cfg(not(armv6m))]
```

Comment:
Right, so we actually need to keep this `not` as: `cfg(all(not(armv6m), feature = "cas"))` to keep `cargo build --target thumbv6m-none-eabi` working.

Reply:
Is that the interface you would like (implicitly masking the features), rather than forcing the user to opt-out with `no-default-features`?

I mentioned this in [this comment](https://github.com/japaric/heapless/pull/120#issuecomment-566554281).

Reply:
To get this PR into a new patch version we need to keep `cargo build --target thumbv6m-none-eabi` (note: "cas" is implicitly enabled) working so the logic needs to make the "cas" feature a no-op on that target.

Reply:
Happy to make the change, though I might suggest that we change the "implicit no-op" behavior on the next breaking (minor) release.

Reply:
And probably make "cas" opt-in as well; I don't like having a crate that by default doesn't build for all targets. Could you make an issue to make sure we don't forget? I'll put on the v0.6.0 milestone.